From 38d7b7cdfde3c50d6263602ffb03af92fcbfa52e Mon Sep 17 00:00:00 2001
From: Calum Lind <calumlind+deluge@gmail.com>
Date: Sat, 13 May 2017 00:05:48 +0100
Subject: [GTKUI] Fix keyerror showing prefs

 * Fix the single_proxy ui to True as unlikely any users using <=0.15
   and need to have different proxy type settings.
---
 deluge/ui/gtkui/preferences.py | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/deluge/ui/gtkui/preferences.py b/deluge/ui/gtkui/preferences.py
index 645ad26..efcd93d 100644
--- a/deluge/ui/gtkui/preferences.py
+++ b/deluge/ui/gtkui/preferences.py
@@ -318,11 +318,8 @@ class Preferences(component.Component):
             }
             # Add proxy stuff
 
-            # Display workaround for single proxy in libtorrent >v0.16
-            try:
-                lt_single_proxy = component.get("PreferencesManager").LT_SINGLE_PROXY
-            except AttributeError:
-                lt_single_proxy = False
+            # Display workaround for single proxy in libtorrent >= v0.16
+            lt_single_proxy = True
 
             for t in ("peer", "web_seed", "tracker", "dht"):
                 if lt_single_proxy and not t == "peer":
-- 
cgit v1.1

